### PR TITLE
Run functional tests from docker image

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
This PR moves our functional-tests into the docker dependency image, allowing for easy test execution with pre-downloaded maven dependencies

The docker entry point script creates a local network and starts che-starter + chromedriver:
`docker network inspect localnetwork`
```
[
    {
        "Name": "localnetwork",
        "Id": "2520bf3332ddee7eba3a670418a07370dc612ca97369888d11abcc7c387dc68e",
        "Created": "2018-10-17T15:21:38.517904145+02:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.21.0.0/16",
                    "Gateway": "172.21.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Containers": {
            "87c96a96cf43814886e919efab1a3153f811a9e98a79ee0730baf73ef1c95365": {
                "Name": "functional-tests-dep",
                "EndpointID": "4267bc0a147d65f4f7966786bed9ef57fa2be5b6e0f807b6b1d5e6c0cf23505d",
                "MacAddress": "02:42:ac:15:00:02",
                "IPv4Address": "172.21.0.2/16",
                "IPv6Address": ""
            },
            "bde691126f8ab72c8ed804b9be24d961da0fde84edcc523be8fb04bb1f9f0a14": {
                "Name": "che-starter",
                "EndpointID": "a7d7221b389f7d8855b152ebf4027e5d841c2355b497be3c1315d63f469af498",
                "MacAddress": "02:42:ac:15:00:03",
                "IPv4Address": "172.21.0.3/16",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {}
    }
]
```

### What issues does this PR fix or reference?
https://github.com/redhat-developer/che-functional-tests/issues/348

### How have you tested this PR?
manual testing

### CODE REVIEW

**Comment by ScrewTSW** [comment]
We need to tag latest build of `quay.io/openshiftio/rhchestage-rh-che-functional-tests-dep` with a static tag before the merge, to allow our old and new test framework to co-exist

@ibuziuk @garagatyi

**Comment by rhopp** [comment]
@ScrewTSW I've tried to build this image locally and run the tests... The tests failed, but not because of this PR. I think we can remove the WIP and merge and the target  the failures in separate issues. WDYT?

File: functional-tests/README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Prerequisities:
 docker run -p 10000:10000 -e "GITHUB_TOKEN_URL=https://auth.openshift.io/api/token?for=https://github.com" -e "OPENSHIFT_TOKEN_URL=https://sso.openshift.io/auth/realms/fabric8/broker/openshift-v3/token" -e "CHE_SERVER_URL=https://che.openshift.io" quay.io/openshiftio/almighty-che-starter:latest
 ```
 
-### Run tests via maven:
+### Run tests via maven
```
Comment (amisevsk) [inline functional-tests/README.md]: Could you add a link to this readme to the [dev guide](https://github.com/redhat-developer/rh-che/blob/master/dev-guide.adoc)

Comment (ScrewTSW) [reply functional-tests/README.md]: will do

**Pull request merged**